src/app/shell/personal-cabinet/shared-cabinet/messages/messages.component.ts

```diff
@@ -0,0 +1,224 @@
+import { WorkshopDeclination } from '../../../../shared/enum/enumUA/declinations/declination';
+import { ChatRoom } from '../../../../shared/models/chat.model';
+import { Workshop } from '../../../../shared/models/workshop.model';
+import { ConfirmationModalWindowComponent } from '../../../../shared/components/confirmation-modal-window/confirmation-modal-window.component';
+import { Constants } from '../../../../shared/constants/constants';
+import { ModalConfirmationType } from '../../../../shared/enum/modal-confirmation';
+import { BlockedParent } from '../../../../shared/models/block.model';
+import { ReasonModalWindowComponent } from '../applications/reason-modal-window/reason-modal-window.component';
+import { CabinetDataComponent } from '../cabinet-data.component';
+import { MatDialog } from '@angular/material/dialog';
+import { NavBarName } from '../../../../shared/enum/navigation-bar';
+import { PushNavPath } from '../../../..//shared/store/navigation.actions';
+import { GetUserChatRooms } from '../../../..//shared/store/chat.actions';
+import { FormControl } from '@angular/forms';
+import { ChatState } from '../../../../shared/store/chat.state';
+import { SearchResponse } from '../../../../shared/models/search.model';
+import { NoResultsTitle } from '../../../../shared/enum/no-results';
+
+@Component({
+  selector: 'app-messages',
+  templateUrl: './messages.component.html',
+  styleUrls: ['./messages.component.scss']
+})
+export class MessagesComponent extends CabinetDataComponent implements OnInit {
+  readonly Role = Role;
+  readonly WorkshopDeclination = WorkshopDeclination;
+  readonly noMessagesTitle = NoResultsTitle.noMessages;
+
+  userRole: Role;
+  providerId: string;
+  filterFormControl: FormControl = new FormControl('');
+  chatRooms: SearchResponse<ChatRoom[]> = { entities: [], totalAmount: 0 };
+
+  @Select(ProviderState.truncated)
+  workshops$: Observable<TruncatedItem[]>;
+  @Select(RegistrationState.provider)
+  provider$: Observable<Provider>;
+  @Select(ChatState.chatRooms)
+  chatRooms$: Observable<SearchResponse<ChatRoom[]>>;
+
+  constructor(protected store: Store, protected matDialog: MatDialog) {
+    super(store, matDialog);
+  }
+
+  protected init(): void {
+    this.userRole = this.store.selectSnapshot<Role>(RegistrationState.role);
+
+    this.provider$
+      .pipe(filter((provider: Provider) => !!provider))
+      .subscribe((provider: Provider) => {
+        this.providerId = provider.id;
+
+        if (this.userRole == Role.provider) {
+          this.getProviderWorkshops();
+        }
+        this.getChats();
+      });
+
+    this.setListeners();
+  }
+
+  protected addNavPath(): void {
+    this.store.dispatch(
+      new PushNavPath({
+        name: NavBarName.Messages,
+        isActive: false,
+        disable: true
+      })
+    );
+  }
+
+  getProviderWorkshops() {
+    this.store.dispatch(new GetWorkshopListByProviderId(this.providerId));
+  }
```

Comment:
it won't support providerAdmin, providerDeputy roles. check how this is implemented in provider-workshops or provider-applications, ALso I think this method can be moved separately to util files (maybe the new one that has only functionality related to the provider roles), because this functionality is implemented more then 2 times